### PR TITLE
motion-triangle

### PR DESCRIPTION
**Pull request by DanieleMaselli**
Awesome you want to make a PR  
please see [https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md](https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md) for further infos.

### CODE REVIEW

File: motion/p5js/motion-triangle/index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli
+title: motion-triangle
+tags: [P5.js, p5.Vector, color, triangle]
+permalink: /color/motion/
+summary: primary colors represented by a moving triangle on P5.js.
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.md]: Please use correct english.

File: motion/p5js/motion-triangle/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli
+title: motion-triangle
+tags: [P5.js, p5.Vector, color, triangle]
+permalink: /color/motion/
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.md]: the permalink can not be `/color/motion/` This is to broad. Please give it a specific name. Be creative. If you commit this into the folder motion the permalink should also start with motion.

File: motion/p5js/motion-triangle/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// create new Vector object
+var movingVector = new p5.Vector(600, 300);
+// create new Vector object
+var changeVector = new p5.Vector(5, 9);
+
+function setup() {
+  var canvas = createCanvas(800, 800);
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: Please use 500*500px

File: motion/p5js/motion-triangle/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  
+  if ( ( movingVector.y > height ) || ( movingVector.y < 0 ) ) {
+    // change sign of v.y (change direction)
+    changeVector.y = changeVector.y * -1;
+  }
+
+  // draws the blue triangle when it is less than 400 on the right side. 
+  if( movingVector.x > 400 ){
+    fill(0, 38, 255); // blue
+    triangle(400, 0, movingVector.x, movingVector.y, 800, 750);
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: Don't use magic numbers

File: motion/p5js/motion-triangle/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    changeVector.x = changeVector.x * -1;
+  }
+  
+  if ( ( movingVector.y > height ) || ( movingVector.y < 0 ) ) {
+    // change sign of v.y (change direction)
+    changeVector.y = changeVector.y * -1;
+  }
+
+  // draws the blue triangle when it is less than 400 on the right side. 
+  if( movingVector.x > 400 ){
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: magic numbers

File: motion/p5js/motion-triangle/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    fill(0, 38, 255); // blue
+    triangle(400, 0, movingVector.x, movingVector.y, 800, 750);
+ 
+    fill(255, 251, 0); // yellow
+    triangle(400, 0, movingVector.x, movingVector.y, 0, 750);
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: magic numbers

File: motion/p5js/motion-triangle/index.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    // ORDER: the blue triangle is under the yellow triangle
+
+  } else {
+    // draw first the yellow triangle
+    fill(255, 251, 0); // yellow
+    triangle(400, 0, movingVector.x, movingVector.y, 0, 750);
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: same here. 🎩 🐇

File: motion/p5js/motion-triangle/index.js (hunk 5 above)
Comment (DanieleMaselli) [inline motion/p5js/motion-triangle/index.js]: ok..lets what i can do

File: motion/p5js/motion-triangle/index.md (hunk 1 above)
Comment (DanieleMaselli) [inline motion/p5js/motion-triangle/index.md]: what wrong about that? sorry not my language ; ) 
primary color expressed by a moving triangle with P5.js? primary color illustrated by a moving triangle with P5.js?

Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.md]: P~~p~~rimary colors represented by a moving triangle in _(or using)_ ~~on~~ P5.js.

Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.md]: Yo should be able to write English. Even if it is not your language.

Comment (DanieleMaselli) [inline motion/p5js/motion-triangle/index.md]: Ok. Thanks

**Comment by DanieleMaselli** [comment]
here we go! code is fixed..now there should be no magic numbers!

File: color/p5js/colored-shapes/index.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  for (var i = 0, side = 30; side > 0; side--, i++) {
+    // fill with color
+    fill(colors[ i % colors.length ], 0, 0);
+    rect( x, y, scaleFactor * side, scaleFactor * side);
+  }
+
+  scaleFactor = 50; 
+  
+  var x = 500;
+	var y = 500;
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-shapes/index.js]: please indent the file properly.

File: color/p5js/colored-shapes/index.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  scaleFactor = 50; 
+  
+  var x = 500;
+	var y = 500; 
+ 	// loop for the radius of the circle
+  for (var i = 0, radius = 14; radius > 0; radius--, i++) {
+  	// fill with color
+ 		fill(0, 0, colors[i % colors.length]);
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-shapes/index.js]: indent

File: color/p5js/colored-shapes/index.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli
+title: colored-shapes
+tags: [P5.js, color, Geometric shapes]
+permalink: /color/shapes/
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-shapes/index.md]: Use a bit more descriptive names.

File: motion/p5js/motion-triangle/index.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  return "rgb(" + color.r + "," + color.g + "," + color.b + ")";
+}
+colors = {
+  blue:   Color({ r: 0, g: 38, b: 255 }) ,
+  yellow: Color({ r: 240, g: 250, b: 0 }) ,
+  red :   Color({ r: 255, g: 0, b: 0 }),
+};
+
+
+var CANVAS_WIDTH  = 500;
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: The canvas width and height are stored in the `width` and `height` variables. If you set the size using `createCanvas` you can access them everywhere in your code. No need to duplicate these with your own variables.

File: motion/p5js/motion-triangle/index.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
+function Color(color){
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: Is this really necessary? you can just use the values. Seems a bit overcomplicated. Also P5.js has already a color function. See --> http://p5js.org/reference/#/p5/color

File: motion/p5js/motion-triangle/index.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+var CANVAS_WIDTH  = 500;
+var CANVAS_HEIGHT = 500;
+
+
+// create new Vector object
+var movingVector = new p5.Vector(100, 100);
+// create new Vector object
+var changeVector = new p5.Vector(5, 10);
+
+scaleFactor = {
```
Comment (fabianmoronzirfas) [inline motion/p5js/motion-triangle/index.js]: please declare variables.  

    var scaleFactor …

File: motion/p5js/motion-triangle/index.js (hunk 11 above)
Comment (DanieleMaselli) [inline motion/p5js/motion-triangle/index.js]: it's a choice of style..

Comment (DanieleMaselli) [inline motion/p5js/motion-triangle/index.js]: it's a choice of style..

**Comment by DanieleMaselli** [comment]
now it should be ok..let me know

File: color/p5js/colored-shapes/index.md (hunk 10 above)
Comment (DanieleMaselli) [inline color/p5js/colored-shapes/index.md]: for tags and link? mmm..

**Comment by fabianmoronzirfas** [comment]
Looks good. Thank you.

**Comment by DanieleMaselli** [comment]
🚀 yeah

**Comment by fabianmoronzirfas** [comment]
Examples are live https://interface.fh-potsdam.de/gestalten-in-code/authors/

**Comment by DanieleMaselli** [comment]
nice!

**Pull request merged**